### PR TITLE
(PC-6698) modifications selection date

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6698

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] |
| -------------------- | 
| image |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/bookOffer/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -51,7 +51,10 @@ const renderDay = (
   if (status === OfferStatus.BOOKABLE)
     return (
       <DayContainer>
-        <TouchableOpacity activeOpacity={ACTIVE_OPACITY} onPress={selectDate}>
+        <TouchableOpacity
+          activeOpacity={ACTIVE_OPACITY}
+          onPress={selectDate}
+          hitSlop={{ top: 8, bottom: 8, left: 8, right: 8 }}>
```
Comment (antoinewg) [inline src/features/bookOffer/components/Calendar/Calendar.tsx:57]: tu peux le définir en constante en dehors du composant (protection pour les perfs)

Comment (emmajmn) [reply src/features/bookOffer/components/Calendar/Calendar.tsx:57]: ok !

File: src/features/bookOffer/components/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -42,6 +42,7 @@ const renderDay = (
     dispatch({ type: 'SELECT_DATE', payload: new Date(date.timestamp) })
     debouncedDispatch({ type: 'CHANGE_STEP', payload: Step.HOUR })
   }
+  const hitSlop = { top: 8, bottom: 8, left: 8, right: 8 }
```
Comment (antoinewg) [inline src/features/bookOffer/components/Calendar/Calendar.tsx:45]: carrément en dehors

**Review by antoinewg — APPROVED** [review]

**Review by antoinewg — CHANGES_REQUESTED** [review]
Il manque des tests. Je pense qu'on va commencer à en mettre, vu que le reducer commence à être stable

File: src/features/bookOffer/pages/BookingOfferModal.tsx [hunk 3]
```diff
@@ -37,7 +41,7 @@ const BookingOfferModalComponent: React.FC<Props> = ({ visible, dismissModal, of
       leftIcon={leftIcon}
       onLeftIconPress={onLeftIconPress}
       rightIcon={Close}
-      onRightIconPress={dismissModal}>
+      onRightIconPress={dismissAndResetModal}>
```
[the remark this answers is not in the document]
Comment (emmajmn) [reply src/features/bookOffer/pages/BookingOfferModal.tsx:44]: ok je modifie ça

File: src/features/bookOffer/pages/__tests__/BookingOfferModal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { fireEvent, render } from '@testing-library/react-native'
+import React from 'react'
+
+import { BookingOfferModalComponent } from '../BookingOfferModal'
+import { mockOffer } from 'features/bookOffer/fixtures/offer'
+import { Step } from '../reducer'
+import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
```
[the remark this answers is not in the document]
Comment (emmajmn) [reply src/features/bookOffer/pages/__tests__/BookingOfferModal.test.tsx:7]: ok solve

File: src/features/bookOffer/pages/__tests__/BookingOfferModal.test.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+describe('<BookingOfferModalComponent />', () => {
+  it('should dismiss modal when click on rightIconButton and reset state', () => {
+    const page = render(
+        reactQueryProviderHOC(<BookingOfferModalComponent visible={true} dismissModal={mockDismissModal} offerId={20} />)
+    )
+
+    const dismissModalButton = page.getByTestId('rightIconButton')
+
+    fireEvent.press(dismissModalButton)
```
[the remark this answers is not in the document]
Comment (emmajmn) [reply src/features/bookOffer/pages/__tests__/BookingOfferModal.test.tsx:27]: pas de warning en vu :)

**Review by antoinewg — APPROVED** [review]

**Pull request merged**